core/trace_point.rbs

```diff
@@ -257,7 +257,7 @@ class TracePoint < Object
   #       C.foo
   #     end
   #
-  def defined_class: () -> Module
+  def defined_class: () -> (Class | Module)?
```

Comment:
Even though `Class | Module` is technically `Module`, i think this is much clearer for the end-user

Reply:
Ok. Give the type a try.